City-of-Helsinki/drupal-module-helfi-tunnistamo
# UHF-5265: Support new Azure AD login

## How to install

- `composer require drupal/helfi_tunnistamo:dev-UHF-5265` 
- `drush updb `
- `drush cr`
- Create a new openid connect client `conf/cmi/openid_connect.client.azure-ad.yml`:
```yaml
uuid: 904cffac-ddb5-4730-ab5f-c41ee12a03b7
langcode: en
status: true
dependencies:
  module:
    - helfi_tunnistamo
id: azure-ad
label: 'Tunnistamo (Azure AD)'
plugin: tunnistamo
settings:
  client_id: ''
  client_secret: ''
  iss_allowed_domains: ''
  auto_login: 0
  environment_url: ''
  client_scopes: 'openid,email'
  client_roles: {  }
```
- Run `drush cim` to import the client
- Add required client configuration to your `local.settings.php` file: 
 ```php
$config['openid_connect.client.azure-ad']['settings']['client_id'] = 'helfi-kaupunkiymparisto-liikenne-developer-dev';
$config['openid_connect.client.azure-ad']['settings']['client_secret'] = ''; // PM me on slack
$config['openid_connect.client.azure-ad']['settings']['ad_roles'] = [
  [
    'ad_role' => 'sg_KANSLIA_Kayttajat',
    'roles' => ['super_administrator'],
  ],
];
$config['openid_connect.client.azure-ad']['settings']['environment_url'] = 'https://tunnistus.test.hel.ninja/auth/realms/helsinki-tunnistus';
```

## How to test

- Go to `/user` and log-in using `Tunnistamo (Azure AD)` client and make sure the user is granted `super_administrator` role automatically.
